addons/point_of_sale/models/pos_session.py

```diff
@@ -23,7 +23,7 @@ def _confirm_orders(self):
             for order in session.order_ids.filtered(lambda o: o.state != 'done'):
                 if order.state not in ('paid', 'invoiced'):
                     raise UserError(_("You cannot confirm all orders of this session, because they have not the 'paid' status"))
-                order.signal_workflow('done')
+                order.action_pos_order_done()
```

Comment:
Thanks for commenting here @celm1990

Indeed, it looks like orders in state 'invoiced' shouldn't get set to 'done' but remain 'invoiced' when closing the session. Seems a bit odd to me, but I'll fix it.